express-notes-api/index.js

```diff
@@ -0,0 +1,109 @@
+const express = require('express');
+const fs = require('fs');
+const app = express();
+let data = null;
+fs.readFile('./data.json', 'utf-8', (err, dataJSON) => {
+  if (err) {
+    process.exit(1);
+  }
+  data = JSON.parse(dataJSON);
+});
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const { notes } = data;
+  let notesArr = [];
+  for (let note in notes) {
+    notesArr.push(notes[note]);
+  }
+  res.status(200).json(notesArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const { notes } = data;
+  const id = req.params.id;
+  if (Number(id) < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+    return;
+  }
+
+  if (notes[id]) {
+    res.status(200).json(notes[id]);
+  } else {
+    res.status(400).json({ error: `cannot find note with id ${id}` });
+  }
```

Comment:
Non-existent (but valid) `id`s should produce a `404 Not Found`:

---

![image](https://user-images.githubusercontent.com/16143771/144651654-021b49a5-3239-4a4a-b919-77786c1df03b.png)